src/taipy/core/pipeline/_pipeline_manager.py

```diff
@@ -124,6 +126,8 @@ def _hard_delete(cls, pipeline_id: PipelineId):
     def _get_owned_entity_ids(cls, pipeline: Pipeline) -> _EntityIds:
         entity_ids = _EntityIds()
         for task in pipeline.tasks.values():
+            if not isinstance(task, Task):
```

Comment:
I remember pipeline.tasks being task ids instead of task entity, if so can we remove the if condition here and go straight to _get?

Reply:
```python   
 @tasks.setter  # type: ignore
    @_self_setter(_MANAGER_NAME)
    def tasks(self, tasks: Union[List[TaskId], List[Task]]):
        self._tasks = tasks
```

This is the signature of pipeline.tasks so we should expect an item can be a task id or a task.

Reply:
```python
        tasks: Union[List[TaskId], List[Task], List[Union[TaskId, Task]]],
```

This is the type annotation of pipeline.tasks so we should expect an item to be either a task id or task.